org.eclipse.xtext.xbase.testing/src/org/eclipse/xtext/xbase/testing/CompilationTestHelper.java

```diff
@@ -147,6 +150,17 @@ public void setJavaCompilerClassPath(ClassLoader classLoader) {
 		this.javaCompiler = new OnTheFlyJavaCompiler2(classLoader, generatorConfigProvider.get(null).getJavaSourceVersion());
 		this.classpathUriContext = classLoader;
 	}
+
+	/**
+	 * Sets the Java version both for the DSL generator, for example, Xbase
+	 * compiler, and for the Java compiler.
+	 * 
+	 * @since 2.11
+	 */
+	public void setJavaVersion(JavaVersion javaVersion) {
+		this.javaCompiler.setJavaVersion(javaVersion);
+		generatorConfigProvider.get(null).setJavaSourceVersion(javaVersion);
```

Comment:
calling the provider with null looks odd. Looking at the implementation it will return a fresh GeneratorConfig with no association.